src/LeanCode.Contracts.Admin/AdminQuery.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace LeanCode.Contracts.Admin;
+
+public abstract class AdminQuery<TResult> : IQuery<AdminQueryResult<TResult>>
+{
+    public int Page { get; set; }
+    public int PageSize { get; set; }
+
+    public SortOrderDTO? SortOrder { get; set; }
+    public string? SortBy { get; set; }
+}
+
+public enum SortOrderDTO
+{
+    Descending = 0,
+    Ascending = 1
+}
+
+public class AdminQueryResult<TResult>
+{
+    public long Total { get; set; }
+    public List<TResult> Items { get; set; }
+}
+
+public class AdminFilterRange<T>
+{
+    public T? From { get; set; }
+    public T? To { get; set; }
+}
+
+[AttributeUsage(AttributeTargets.Property | AttributeTargets.Field, AllowMultiple = false)]
+public class AdminFilterFor : Attribute
+{
+    public AdminFilterFor(string name) { }

```

Comment:
Won't the value of `name` be lost after construction? You should probably preserve it in a field/property.

Same in attributes below.